adapters/rubicon/rubicon.go

```diff
@@ -387,14 +387,20 @@ func (a *RubiconAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *
 			rubiReq.App = &appCopy
 		}
 
-		err = json.NewEncoder(&requests[i]).Encode(rubiReq)
+		var reqBuffer bytes.Buffer
+		err = json.NewEncoder(&reqBuffer).Encode(rubiReq)
+		requestsJSON = append(requestsJSON, reqBuffer)
 		if err != nil {
 			return nil, err
 		}
 	}
+	if len(requestsJSON) == 0 {
+		return nil, errors.New("Invalid ad unit size")
+	}
 
 	ch := make(chan adapters.CallOneResult)
-	for i := range bidder.AdUnits {
+
+	for _, requestJSON := range requestsJSON {
```

Comment:
You may want to use another name instead of "requestJSON", "requestJSON" and "requestsJSON" look similar to each other..

Reply:
similar to something like `for bidder := range bidders`,  I used it because it is the same thing. 
I'll change it if it's bad practice.

Reply:
This's my personal opinion, and it's a local variable (not widely used). Should be fine.

Reply:
BTW, this is a really good update which can increase our bid requests to XAPI